osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/DlmsObjectConfigService.java

```diff
@@ -0,0 +1,196 @@
+            final List<DlmsCaptureObject> selectedObjects) {
+        return this.findDlmsObject(Protocol.withNameAndVersion(device.getProtocol(), device.getProtocolVersion()), type,
+                filterMedium)
+                .map(dlmsObject -> new AttributeAddress(dlmsObject.getClassId(),
+                        this.replaceChannel(dlmsObject.getObisCode(), channel), dlmsObject.getDefaultAttributeId(),
+                        this.getAccessDescription(dlmsObject, from, to, channel, filterMedium, device,
+                                selectedObjects)));
+    }
+
+    private Optional<DlmsObject> findDlmsObject(final Protocol protocol, final DlmsObjectType type,
+            final Medium filterMedium) {
+        return this.dlmsObjectConfigs.stream()
+                .filter(config -> config.contains(protocol))
+                .findAny()
+                .flatMap(dlmsObjectConfig -> this.findDlmsObject(dlmsObjectConfig, type, filterMedium));
+    }
+
+    private Optional<DlmsObject> findDlmsObject(final DlmsObjectConfig dlmsObjectConfig, final DlmsObjectType type,
+            final Medium filterMedium) {
+        // @formatter:off
+        return dlmsObjectConfig.getObjects()
+                .filter(o1 -> o1.getType().equals(type))
+                .filter(o2 -> !(o2 instanceof DlmsProfile)
+                        || ((DlmsProfile) o2).getMedium() == Medium.COMBINED
+                        || ((DlmsProfile) o2).getMedium() == filterMedium)
+                .findAny();
+        // @formatter:on
+    }
+
+    public List<AttributeAddress> getAttributeAddressesForScalerUnit(final List<DlmsCaptureObject> selectedObjects,
+            final Integer channel) {
+        final List<AttributeAddress> attributeAddresses = new ArrayList<>();
+
+        // Get all Registers from the list of selected objects for which the default attribute is captured.
+        final List<DlmsRegister> dlmsRegisters = selectedObjects.stream()
+                .filter(c -> c.getAttributeId() == c.getRelatedObject().getDefaultAttributeId())
+                .map(DlmsCaptureObject::getRelatedObject)
+                .filter(r -> r instanceof DlmsRegister)
+                .map(r -> (DlmsRegister) r)
+                .collect(Collectors.toList());
+
+        for (final DlmsRegister register : dlmsRegisters) {
+            attributeAddresses.add(
+                    new AttributeAddress(register.getClassId(), this.replaceChannel(register.getObisCode(), channel),
+                            register.getScalerUnitAttributeId()));
+        }
+
+        return attributeAddresses;
+    }
+
+    private ObisCode replaceChannel(String obisCode, final Integer channel) {
+        if (channel != null) {
+            obisCode = obisCode.replace("<c>", channel.toString());
+        }
+
+        return new ObisCode(obisCode);
+    }
+
+    private SelectiveAccessDescription getAccessDescription(final DlmsObject object, final DateTime from,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 7 parameters, which is greater than 6 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00107)